compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -856,10 +855,32 @@ impl<'a, 'b> BuildReducedGraphVisitor<'a, 'b> {
         } else if orig_name == Some(kw::SelfLower) {
             self.r.graph_root
         } else {
-            let crate_id =
-                self.r.crate_loader.process_extern_crate(item, &self.r.definitions, local_def_id);
-            self.r.extern_crate_map.insert(local_def_id, crate_id);
-            self.r.expect_module(crate_id.as_def_id())
+            match self.r.crate_loader.process_extern_crate(item, &self.r.definitions, local_def_id)
+            {
+                Some(crate_id) => {
+                    self.r.extern_crate_map.insert(local_def_id, crate_id);
+                    self.r.expect_module(crate_id.as_def_id())
+                }
+                _ => {
+                    let dummy_import = self.r.arenas.alloc_import(Import {
```

Comment:
The only difference between the dummy and non-dummy import is `imported_module`, so you can do something like
```rust
let binding = match self.r.crate_loader.process_extern_crate(item, &self.r.definitions, local_def_id) {
  Some(crate_id) => { / * non-dummy binding */ }
  None => { / * dummy binding */ }
}

let import = self.r.arenas.alloc_import(Import { ... });
```